.github/workflows/cpp_windows.yml

```diff
@@ -36,15 +36,18 @@ on:
 jobs:
   windows:
     runs-on: ${{ inputs.os }}
-    timeout-minutes: 60
+    timeout-minutes: 240
     env:
-      ARROW_BOOST_USE_SHARED: OFF
       ARROW_BUILD_BENCHMARKS: ON
       ARROW_BUILD_SHARED: ON
-      ARROW_BUILD_STATIC: OFF
+      ARROW_BUILD_STATIC: ON
       ARROW_BUILD_TESTS: ON
+      ARROW_BUILD_TYPE: release
```

Comment:
TODO - raise an issue for this: Changed build type to `release`, because I was getting `error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in unity_2_cxx.cxx.obj` error in `debug` build. 
Run log: https://github.com/apache/arrow/actions/runs/19152083026/job/54744585087

The issue seems to be due to vcpkg dependencies built in release but the build itself is in debug. The `vcpkg` dependencies are not automatically built in debug